File: app/src/main/java/e/dell_user/finalproject_1/MainActivity.java
package e.dell_user.finalproject_1;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    private Button mBtnShowResult;

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //取得此Button的實體
        mBtnShowResult = (Button)findViewById(R.id.btn_score);

        //實做OnClickListener界面
        mBtnShowResult.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //初始化Intent物件
                Intent intent = new Intent(getBaseContext(),RunResultActivity.class);
                //從MainActivity 到Main2Activity
                //intent.setClass(MainActivity.this , ResultActivity.class);
                //開啟Activity
                startActivity(intent);
            }
        });
    }

}

File: app/src/main/java/me/yaoandy107/androidhw7/MainActivity.kt
package me.yaoandy107.androidhw7

import android.content.Intent
import android.graphics.drawable.AnimationDrawable
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private var userWin: Int = 0
    private var userDraw: Int = 0
    private var userLose: Int = 0
    private var count:Int = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_draw.setOnClickListener {
            drawDice()
        }
        btn_score.setOnClickListener {
            val intent = Intent(this, ResultActivity::class.java)
            val bundle = Bundle()
            bundle.putString("count", count.toString())
            bundle.putString("win", userWin.toString())
            bundle.putString("draw", userDraw.toString())
            bundle.putString("lose", userLose.toString())
            intent.putExtras(bundle)
            startActivity(intent)
        }
    }

    private fun drawDice() {
        val animationDrawable = iv_dice.drawable as AnimationDrawable
        animationDrawable.start()
        Handler().postDelayed({
            count++
            val random = (Math.random() * 6 + 1).toInt()
            when (random) {
                1, 2 -> {
                    userWin++
                    Toast.makeText(baseContext, "玩家贏", Toast.LENGTH_SHORT).show()
                }
                3, 4 -> {
                    userDraw++
                    Toast.makeText(baseContext, "平手", Toast.LENGTH_SHORT).show()
                }
                5, 6 -> {
                    userLose++
                    Toast.makeText(baseContext, "電腦贏", Toast.LENGTH_SHORT).show()
                }
            }
            animationDrawable.stop()
            animationDrawable.selectDrawable(random - 1)
        }, 1000)
    }
}
